schematic/base.py

```diff
@@ -0,0 +1,446 @@
+                return cls(parameter=cls.get_parameter_for_value(value))
+        else:
+            return cls()
+
+
+class TableColumn(ABC, DictableMixin, NameSqlMixin):
+    """DB-agnostic base class for storing info about a column in a table.
+
+    Attributes:
+      name: The identifier for the column
+      type: A TableColumnType instance with type information for the column.
+    """
+
+    def __init__(self, name, column_type):
+        self.name = name
+        self.column_type = column_type
+
+    def __hash__(self):
+        return hash((self.name, self.column_type))
+
+    def __repr__(self):
+        return "{}: {}".format(self.name, self.column_type)
+
+    def __eq__(self, other):
+        return isinstance(self, type(
+            other)) and self.name == other.name and self.column_type == other.column_type
+
+
+class TableDefinition(ABC, DictableMixin, NameSqlMixin):
+    """DB-agnostic base class for storing info about a table
+
+    Attributes:
+      name: The identifier for this table.
+      columns: A list of TableColumns describing this table's columns.
+    """
+
+    def __init__(self, name, columns):
+        self.name = name
+        self.columns = columns
+
+    def create_sql(self):
+        """Generate a sql statement for creating a table based on this TableDefinition.
+
+        Raises:
+          NotImplementedError: Subclasses should implement this.
+        """
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        return isinstance(
+            self,
+            type(other)) and self.name == other.name and frozenset(
+            self.columns) == frozenset(
+            other.columns)
+
+    def __repr__(self):
+        return "{}: [{}]".format(self.name, ",".join(
+            [str(col) for col in self.columns]))
+
+    def create_table(self, *args, **kwargs):
+        """Create the table in the destination
+            specified in *args and **kwargs
+
+        Raises:
+          NotImplementedError: Subclasses should implement this.
+        """
+        raise NotImplementedError
+
+    def add_column(self, column):
+        """Add a column to the column list.
+
+        Args:
+          column: A TableColumn instance to be added to this instance's columns.
+        """
+        for col in self.columns:
+            if col.name.upper() == column.name.upper():
+                raise ValueError(
+                    "Column with name {} already exists in TableDefinition {}".format(
+                        column.name, self.name))
+        self.columns.append(column)
+
+    def update_column(self, column):
+        """Update the existing column with name column.name to the given column.
+
+        Args:
+          column: A TableColumn instance to be added to this instance's columns.
+        Raises:
+          ValueError: If TableColumn with column.name does not exist in columns
+        """
+        for i, col in enumerate(self.columns):
+            if col.name == column.name:
```

Comment:
```suggestion
            if col.name.upper() == column.name.upper():
```
to be consistent with above equality logic

Reply:
If you take my earlier advice and just store column.name in uppercase, disregard